File: 1475.final-prices-with-a-special-discount-in-a-shop.java
// @leet start
class Solution {
	public int[] finalPrices(int[] prices) {
		int[] answer = new int[prices.length];
		for (int i = 0; i < prices.length; i++) {
			int j = i + 1;
			while (j < prices.length && prices[j] > prices[i])
				j++;
			if (j == prices.length) {
				answer[i] = prices[i];
			} else {
				answer[i] = prices[i] - prices[j];
			}
		}
		return answer;
	}
}
// @leet end

File: 1475.final-prices-with-a-special-discount-in-a-shop.py
# @leet start
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        answer = []
        for i in range(len(prices)):
            j = i + 1
            while j < len(prices) and prices[i] < prices[j]:
                j += 1
            if j == len(prices):
                answer.append(prices[i])
            else:
                answer.append(prices[i] - prices[j])
        return answer


# @leet end
